documentation/directory-structure.md

```diff
@@ -0,0 +1,201 @@
+but of a **service group**. A service group combines multiple resource providers,
+from multiple teams, in one `specification/rp` folder. In this situation the layout
+constraints described in this article are partially violated, e.g. given `rpnamespace` instead
+of having `stable` and `preview` as direct children, has extra layer of "sub-service" name, like `rpnamespace/compute`. This scenario is not allowed going forward and existing teams will migrate away from it, eventually.
+
+Examples of an existing services groups: 
+- [`specification/compute`]
+- [`specification/workloads`]
+
+## common-types
+
+Specification files and AutoRest configuration files in one resource provider folder are better to refer to files in the same resource provider folder. Entity type definition that can be shared cross resource providers or services should to be placed and maintained either under the folder [**common-types**](https://github.com/Azure/azure-rest-api-specs#common-types) under specification, or under **common-types** folder of service group folder structure. The former supports the entity type sharing cross rp folders, while the latter supports the entity type sharing cross components or services under the same rp folder.
+
+Refer to [Resource-Management](https://github.com/Azure/azure-rest-api-specs/tree/main/specification/common-types/resource-management) common types for example.
+
+## FAQ
+
+### Should my service team have one or multiple resource provider folders?
+
+Should given service team add multiple `specification/rp` folders, or just one?
+It depends on the following considerations:
+    
+- An rp folder leads to a separate SDK package. Is it expected to have separate SDK packages for different services?
+- Service in one folder share the same versioning cycle. Can service in one folder share the same version label, and upgrade together in the future?
+- Specification files and AutoRest configuration files in one rp folder are better to refer to files in the same rp folder. Note: Entity type definition that needs to be referred cross rp folders should be placed and maintained under the folder [**common-types**](https://github.com/Azure/azure-rest-api-specs#common-types).
+- For more considerations, you may consult the reviewer in API design review. To initiate the review, Please submit an [Azure SDK intake questionnaire](https://aka.ms/sdk-apex).
+
+[aka.ms/azsdk/pr-api-versions]: https://aka.ms/azsdk/pr-api-versions
+[aka.ms/azsdk/typespec]: https://aka.ms/azsdk/typespec
+[aka.ms/typespec]: https://aka.ms/typespec
+[resource provider list]: https://learn.microsoft.com/en-us/azure/azure-resource-manager/management/azure-services-resource-providers#match-resource-provider-to-service
+[RP Onboarding]: https://armwiki.azurewebsites.net/rp_onboarding/process/onboarding.html#0-on-boarding-meeting
+[TypeSpec directory structure]: https://github.com/Azure/azure-rest-api-specs/blob/main/documentation/typespec-structure-guidelines.md
```

Comment:
We should consider deleting https://github.com/Azure/azure-rest-api-specs/blob/main/documentation/typespec-structure-guidelines.md and folding that information inot this document.

Reply:
I'm happy with doing this later but I would like to get this added in here.

Reply:
Yeah, let's do it in a follow-up PR.

Reply:
Filed an issue for that:
- https://github.com/Azure/azure-sdk-tools/issues/8028